common/djangoapps/third_party_auth/api/urls.py

```diff
@@ -0,0 +1,12 @@
+""" URL configuration for the third party auth API """
+
+from django.conf.urls import patterns, url
+
+from .views import UserView
+
+USERNAME_PATTERN = r'(?P<username>[\w.+-]+)'
```

Comment:
Hmm, I agree but that will require changes to the CMS, LMS, `common`, and `openedx` modules, and will make this diff harder to review. Since we're pressed for time here, I think I'm going to let that go for now. This would be a nice thing to refactor at the same time as adding a way to share config/constants between the CMS and LMS. Perhaps in the `openedx` module.